### DISCUSSION TITLE
How does the Touch Button work (in the code)

### DISCUSSION THREAD

**Discussion by ghxstweb** [Schneegans/Fly-Pie#336] [Fly-Pie Q&A]
Hi Simon! May I ask how you set up the touch button in FlyPie to display over all over the windows? I noticed that despite the fact that the icon hovers over all of the windows, I am still able to right click on the main desktop background, and, generally, none of the other windows below it are affected by the icon other than the space in which it occupies. How did you configure it to work so well? I'd appreciate any links to the code if possible! Thanks again for your work on this app. It's very versatile and I'm able to use the Touch button very effectively with touchscreen devices...

**Comment by Schneegans** [comment]
Hi there! The buttons are created in [this method](https://github.com/Schneegans/Fly-Pie/blob/main/src/extension/TouchButtons.js#L236). With GNOME Shell extensions, it's pretty straight-forward to create UI elements which are drawn above everything (Fly-Pie itself is another example). The button actors are added to the Shell [here](https://github.com/Schneegans/Fly-Pie/blob/main/src/extension/TouchButtons.js#L450).

Does this help you already?
